cmds/cpu/cpu9p_unix_test.go

```diff
@@ -197,3 +197,146 @@ func Test9pUnix(t *testing.T) {
 	}
 
 }
+
+func Test9pUnixTable(t *testing.T) {
+	d := t.TempDir()
+	f := filepath.Join(d, "a")
+	if err := ioutil.WriteFile(f, []byte("hi"), 0666); err != nil {
+		t.Fatalf(`ioutil.WriteFile(%q, "hi", 0666): %v != nil`, f, err)
+	}
+	// First test: do nothing
+	m := p9.AttrMask{
+		Mode:        true,
+		NLink:       true,
+		UID:         true,
+		GID:         true,
+		RDev:        true,
+		ATime:       true,
+		MTime:       true,
+		CTime:       true,
+		INo:         true,
+		Size:        true,
+		Blocks:      true,
+		BTime:       true,
+		Gen:         true,
+		DataVersion: true,
+	}
+
+	c := &cpu9p{
+		path: f,
+	}
+
+	_, _, ga, err := c.GetAttr(m)
+	if err != nil {
+		t.Fatalf("Getattr: %v != nil", err)
+	}
+
+	sam := p9.SetAttrMask{
+		Permissions:        false,
+		UID:                false,
+		GID:                false,
+		Size:               false,
+		ATime:              false,
+		MTime:              false,
+		CTime:              false,
+		ATimeNotSystemTime: false,
+		MTimeNotSystemTime: false,
+	}
+	sa := p9.SetAttr{
+		Permissions:      0777,
+		UID:              ga.UID,
+		GID:              ga.GID,
+		Size:             ga.Size + 4,
+		ATimeSeconds:     ga.ATimeSeconds + 22,
+		ATimeNanoSeconds: ga.ATimeNanoSeconds + 33,
+		MTimeSeconds:     ga.MTimeSeconds + 55,
+		MTimeNanoSeconds: ga.MTimeNanoSeconds + 66,
+	}
+
+	var tests = []string{
+		"Change nothing",
+		"Setattr with Permissions",
+		"Setattr with size",
+		"Setattr with UID",
+		"Setattr with GID",
+		"Setattr with ATime to local system time",
+		"Setattr with MTime to local system time",
+		"Settattr with ATime to time in TSetattr",
+		"Settattr with MTime to time in TSetattr",
+	}
+	for i, msg := range tests {
+		switch i {
```

Comment:
This should be a ton easier. Now the test has logic for data; that'd be the point though, to separate the two.

Can we create a function doing something like "give me a modified sam with `Permissions` set to `true`" in Go?

Reply:
Well, and that function should accept any modification.
Can we pass a function that the function executes, doing said modification?

Reply:
In other words: I want the test suite to look just like data.

Data defining what each case means; can we put those switch statements into `tests`?

Reply:
Reading through https://medium.com/@geisonfgfg/functional-go-bc116f4c96a4#:~:text=Golang%20is%20a%20multi%2Dparadigm,easier%20to%20understand%20and%20test. suggests that Go lacks the necessary primitives.

I'm sorry, I was hoping that'd be easier. It would be possible to hack, but that'd only make things worse, I think.